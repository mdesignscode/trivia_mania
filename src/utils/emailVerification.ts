import { env } from '$env/dynamic/private';
import { EmailVerification } from '../models';
import nodemailer from 'nodemailer';

const sender = env.EMAIL_ADDRESS;
const pass = env.EMAIL_CODE;

// Configure the transporter
const transporter = nodemailer.createTransport({
        host: 'smtp.gmail.com', // e.g. smtp.gmail.com
        port: 587,
        secure: false, // true for 465, false for other ports
        auth: {
                user: sender,
                pass,
        },
});

// Send email function
export const sendEmail = async ({ email, username }: { username: string, email: string }) => {
        // generate email verification code
        const code = await generateEmailVerificationCode(email);

        // email body
        const textBody = `<p>Hi ${username},</p>
    <p>Thank you for signing up with <strong>Trivia Mania</strong>! To complete your registration, please use the verification code below:</p>
    <div class="code-box">${code}</div>
    <p>This code will expire in <strong>10 minutes</strong>. If you did not request this, you can safely ignore this email.</p>
    <p>Need help? <a href="mailto:${sender}">Contact our support team</a>.</p>
    <div class="footer">
      &copy; 2025 Trivia Mania. All rights reserved.
      <br>
      <a href="https://triviamania.vercel.app">https://triviamania.vercel.app</a>
    </div>`

        try {
                // send email
                await transporter.sendMail({
                        from: '"Trivia Mania" ejue1vs7n@mozmail.com', // sender address
                        to: email, // list of receivers
                        subject: 'Verify Your Email Address', // Subject line
                        html: textBody, // HTML body
                });

                return { status: 'success', data: code };
        } catch (error) {
                console.error('Error sending email:', error);
                return { status: 'fail', data: error.message };
        }
};


export async function generateEmailVerificationCode(email: string): Promise<string> {
        const code = Math.floor(100000 + Math.random() * 900000).toString(); // 6-digit code
        const expiresAt = new Date(Date.now() + 10 * 60 * 1000); // 10 minutes from now

        // existing verification should be updated
        const existingVerification = await EmailVerification.findOne({
                where: { email },
        });

        if (existingVerification) {
                existingVerification.setDataValue('code', code);
                existingVerification.setDataValue('expiresAt', expiresAt);
                await existingVerification.save();
                return code;
        }

        await EmailVerification.create({
                email,
                code,
                expiresAt,
                used: false,
        });

        return code;
}

export async function validateEmailVerificationCode(email: string, inputCode: string): Promise<{ success: boolean; message: string }> {
        const record = await EmailVerification.findOne({
                where: {
                        email,
                },
        });

        const failedResponse = { success: false, message: 'Invalid or expired verification code.' };

        if (record) {
                // if code hasn't expired yet
                const codeExpired = record.getDataValue('expiresAt') < new Date();
                if (codeExpired) return failedResponse;

                // is code correct
                const code = record.getDataValue('code');
                if (code !== inputCode) return failedResponse;

                // verified
                record.setDataValue('used', true)
                await record.save();

                return { success: true, message: 'Verification successful.' };
        }

        // record not found
        return failedResponse;
}

