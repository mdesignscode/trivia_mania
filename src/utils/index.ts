import { browser } from '$app/environment';

export function decodeHTMLEntities(text: string) {
        if (browser) {
                const element = document.createElement("div");
                element.innerHTML = text;
                return element.textContent || "";
        }
}

export const colorMap: { [key: string]: string } = {
        easy: "green",
        medium: "gold",
        hard: "red",
};

export function capitalize(str: string): string {
	if (!str) return '';
	return str.charAt(0).toUpperCase() + str.slice(1).toLowerCase();
}

