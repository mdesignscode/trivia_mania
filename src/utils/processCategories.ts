export type TQuestionData = {
        category: string;
        difficulty: string;
        count: number;
};
export type TAllCategories = Record<string, Record<string, TQuestionData>>;

export type TProcessResult = {
        allCategories: TAllCategories;
        allDifficulties: string[];
};

export function processCategories(
        categories: TQuestionData[],
        difficulties: string[]
): TProcessResult {
        const allCategories: TAllCategories = {};

        // map difficulties to categories
        categories.forEach(({ difficulty, count, category }) => {
                if (!allCategories[category]) allCategories[category] = {};
                allCategories[category][difficulty] = { difficulty, count, category };
        });

        // add "all difficulties" count for each category
        Object.keys(allCategories).forEach((key) => {
                const easy = allCategories[key].easy?.count || 0;
                const medium = allCategories[key].medium?.count || 0;
                const hard = allCategories[key].hard?.count || 0;
                const allCount = easy + medium + hard;

                allCategories[key]['all difficulties'] = {
                        difficulty: 'all difficulties',
                        count: allCount,
                        category: key
                };
        });

        // total "All Categories" aggregate
        const totalCategories: Record<string, TQuestionData> = {
                easy: { difficulty: 'easy', count: 0, category: 'All Categories' },
                medium: { difficulty: 'medium', count: 0, category: 'All Categories' },
                hard: { difficulty: 'hard', count: 0, category: 'All Categories' },
                'all difficulties': {
                        difficulty: 'all difficulties',
                        count: 0,
                        category: 'All Categories'
                }
        };

        Object.values(allCategories).forEach((catGroup) => {
                ['easy', 'medium', 'hard'].forEach((level) => {
                        if (catGroup[level]) {
                                totalCategories[level].count += catGroup[level].count;
                                totalCategories['all difficulties'].count += catGroup[level].count;
                        }
                });
        });

        allCategories['All Categories'] = totalCategories;

        return {
                allCategories,
                allDifficulties: [...difficulties]
        };
}

