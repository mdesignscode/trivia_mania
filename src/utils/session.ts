import { db } from '$lib/db';
import { encodeBase32LowerCaseNoPadding, encodeHexLowerCase } from '@oslojs/encoding';
import { sha256 } from '@oslojs/crypto/sha2';

export function generateSessionToken(): string {
        const bytes = new Uint8Array(20);
        crypto.getRandomValues(bytes);
        return encodeBase32LowerCaseNoPadding(bytes);
}

export function createSession(token: string, userId: number) {
        const sessionId = encodeHexLowerCase(sha256(new TextEncoder().encode(token)));
        const expiresAt = new Date(Date.now() + 1000 * 60 * 60 * 24 * 30); // 30 days

        db.prepare(
                'INSERT INTO session (id, user_id, expires_at) VALUES (?, ?, ?)'
        ).run(sessionId, userId, Math.floor(expiresAt.getTime() / 1000));
}

export function validateSessionToken(token: string) {
        const sessionId = encodeHexLowerCase(sha256(new TextEncoder().encode(token)));

        const row = db.prepare(`
    SELECT session.id, session.user_id, session.expires_at, user.username
    FROM session
    INNER JOIN user ON user.id = session.user_id
    WHERE session.id = ?
  `).get(sessionId);

        if (!row) {
                return { session: null, user: null };
        }

        const session = {
                id: row.id as string,
                userId: row.user_id as number,
                expiresAt: new Date(row.expires_at * 1000)
        };
        const user = {
                username: row.username as string
        };

        if (Date.now() >= session.expiresAt.getTime()) {
                db.prepare('DELETE FROM session WHERE id = ?').run(session.id);
                return { session: null, user: null };
        }

        return { session, user };
}

export function invalidateSession(sessionId: string) {
        db.prepare('DELETE FROM session WHERE id = ?').run(sessionId);
}

