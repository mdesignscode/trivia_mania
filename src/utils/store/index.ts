import globalStore from './global.svelte';
import { questionStore, playStore } from './playStore.svelte';

export {
        globalStore, playStore, questionStore,
}

