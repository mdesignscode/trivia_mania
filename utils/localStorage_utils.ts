export const QUESTIONS_POOL = "questionsPool",
  POOL_INDEX = "poolIndex",
  CURRENT_INDEX = "currentIndex",
  QUESTIONS_LIST = "questionsList",
  QUESTION_ANSWERED = "questionAnswered",
  ANSWERED_QUESTIONS = "answeredQuestions",
  PROGRESS = "progress",
  LAST_ANSWER = "lastAnswer",
  LAST_ANSWER_INDEX = "lastAnswerIndex",
  NEW_PARAMS = "newParams",
  CATEGORIES = "categories",
  DIFFICULTY = "difficulty",
  UNSAVED_DATA = "unsavedData",
  USERNAME = "username"

export function clearQuestionData() {
  localStorage.removeItem(UNSAVED_DATA);
  localStorage.removeItem(QUESTIONS_LIST);
  localStorage.removeItem(QUESTIONS_POOL);
  localStorage.removeItem(POOL_INDEX);
  localStorage.removeItem(CURRENT_INDEX);
  localStorage.removeItem(ANSWERED_QUESTIONS);
  localStorage.removeItem(LAST_ANSWER);
  localStorage.removeItem(LAST_ANSWER_INDEX);
  localStorage.removeItem(ANSWERED_QUESTIONS);
  localStorage.removeItem(NEW_PARAMS);
}
