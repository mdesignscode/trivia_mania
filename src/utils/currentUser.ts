import { redirect, type Cookies } from "@sveltejs/kit";
import { validateSessionToken } from "./session";

// check if user is already logged in
export const isSignedIn = async (token: string | undefined, route: string) => {
        // don't redirect auth pages
        const isAuthRoute = ['/login', '/signup'].includes(route);
        if (!token && isAuthRoute) return;

        // user not logged in
        if (!token) throw redirect(302, '/login');

        const { user } = await validateSessionToken(token);
        // user signed in
        if (user) throw redirect(302, '/');

        // no user
        if (!user && isAuthRoute) return;
        else throw redirect(302, '/login');
}

export const getUser = async (cookies: Cookies) => {
        const token = cookies.get('session');
        if (!token) {
                throw redirect(302, '/login');
        }

        const { session, user } = await validateSessionToken(token);
        if (!session || !user) {
                throw redirect(302, '/login');
        }

        const userData = user.dataValues;
        delete userData.password;

        return {
                user: userData,
        };
}

