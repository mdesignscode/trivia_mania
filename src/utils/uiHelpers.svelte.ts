import { globalStore } from 'store';
import type { TAllCategories } from './processCategories';

export function updateLocalPreferences(allCategories: TAllCategories, localStorage: Storage) {
        const localDifficulty = localStorage.getItem('difficulty') || '';
        const localSelectedCategories = localStorage.getItem('selectedCategories') || '';
        const selectedCategories = localSelectedCategories ? localSelectedCategories.split(',') : [];
        const availableCategories: string[] = [];
        let filteredCategories;

        // check availability of category
        selectedCategories.forEach((c) => {
                if (allCategories[c] && allCategories[c][localDifficulty]) {
                        availableCategories.push(c);
                }
        });

        // update local storage with available categories
        if (selectedCategories.some((c) => !availableCategories.includes(c)))
                localStorage.setItem('selectedCategories', availableCategories.join(','));

        // sync filters
        globalStore.difficulty = localDifficulty;
        globalStore.categories = availableCategories;

        if (localDifficulty) {
                filteredCategories = Object.values(allCategories)
                        .map((value) => value[localDifficulty])
                        .filter(Boolean);
        }
        return filteredCategories;
}

export function handleSelectCategory(category: string) {
        let selectedCategories = globalStore.categories.includes(category)
                ? globalStore.categories.filter((cat) => cat !== category)
                : [...globalStore.categories, category];
        if (category === 'All Categories') {
                if (!globalStore.categories.includes(category)) selectedCategories = [category];
        } else {
                if (globalStore.categories.includes('All Categories'))
                        selectedCategories = selectedCategories.filter((cat) => cat !== 'All Categories');
        }
        globalStore.categories = selectedCategories;
};

export function handleSetDifficulty(localStorage: Storage, allCategories: TAllCategories, difficulty?: string) {
        if (difficulty && typeof difficulty === 'string') {
                globalStore.difficulty = difficulty;
                localStorage.setItem('difficulty', difficulty);
        }
        const filteredCategories = Object.values(allCategories)
                .map((value) => value[globalStore.difficulty])
                .filter(Boolean);
        return filteredCategories;
};

export const handlePlay = (localStorage: Storage, goto: (path: string) => {}) => {
        const playUri = encodeURI(
                `/play?difficulty=${globalStore.difficulty}&categories=${globalStore.categories.join(',')}`
        );
        localStorage.setItem('difficulty', globalStore.difficulty);
        localStorage.setItem('selectedCategories', globalStore.categories.join(','));
        goto(playUri);
};

