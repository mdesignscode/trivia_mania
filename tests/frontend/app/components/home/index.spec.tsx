import HomePage from "@/components/home";
import {
  mockCategoryStats,
  mockDifficultyStats,
  mockEasyStats,
  mockHardStats,
  mockMediumStats,
} from "@/utils/mockData";
import {
  mockGetQuestionStats,
  renderHomeContext,
} from "@/utils/test_home_context";
import { screen, userEvent } from "@/utils/test_utils";

// mock axios post to return different stats
jest.mock("axios", () => ({
  post: (url: string, body: any) => {
    if (body.difficulty == null) return { data: mockDifficultyStats };

    const difficulty = body.difficulty;

    switch (difficulty) {
      case "easy":
        return { data: mockEasyStats };

      case "medium":
        return { data: mockMediumStats };

      case "hard":
        return { data: mockHardStats };

      default:
        return { data: mockCategoryStats };
    }
  },
}));

describe("HomePage component", () => {
  it("Renders <Welcome />", async () => {
    const { baseElement } = renderHomeContext(<HomePage />, {
      currentUI: {
        welcome: true,
        difficulties: false,
        categories: false,
        play: false,
      },
    });

    // get child components' containers
    const container = await screen.findByTestId("home-container"),
      welcomeContainer = await screen.findByTestId("welcome-container"),
      continueButton = await screen.findByTestId("welcome-continue-button");

    expect(container).toBeInTheDocument();
    expect(welcomeContainer).toBeInTheDocument();
    expect(continueButton).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it("Renders <Difficulties />", async () => {
    const { baseElement } = renderHomeContext(<HomePage />, {
      currentUI: {
        welcome: false,
        difficulties: true,
        categories: false,
        play: false,
      },
    });

    // get difficulties container
    const difficultiesContainer = await screen.findByTestId(
        "render-difficulties-container"
      ),
      container = await screen.findByTestId("home-container");

    expect(container).toBeInTheDocument();
    expect(difficultiesContainer).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  it("Renders <Difficulties />, <Categories /> and <Play />", async () => {
    const { baseElement } = renderHomeContext(<HomePage />, {
      currentUI: {
        welcome: false,
        difficulties: true,
        categories: true,
        play: true,
      },
      showCategories: true,
    });

    // get child components' containers
    const container = await screen.findByTestId("home-container"),
      categoriesContainer = await screen.findByTestId(
        "render-categories-container"
      ),
      playLink = await screen.findByTestId("play-link"),
      difficultiesContainer = await screen.findByTestId(
        "render-difficulties-container"
      );

    // assert containers are in DOM
    expect(container).toBeInTheDocument();
    expect(difficultiesContainer).toBeInTheDocument();
    expect(categoriesContainer).toBeInTheDocument();
    expect(playLink).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it("Should gracefully handle user interaction", async () => {
    const user = userEvent.setup();

    renderHomeContext(<HomePage />, {
      categoryStats: mockCategoryStats,
      currentUI: {
        welcome: false,
        difficulties: true,
        categories: true,
        play: true,
      },
      showCategories: true,
    });

    // a list of difficulty buttons should be displayed
    for (const key of Object.keys(mockDifficultyStats)) {
      const button = await screen.findByTestId(key);

      expect(button).toBeInTheDocument();
    }

    // a list of category buttons should be displayed
    for (const key of Object.keys(mockCategoryStats)) {
      const button = await screen.findByTestId(key);

      expect(button).toBeInTheDocument();
    }

    const easy = await screen.findByTestId("easy");

    await user.click(easy);

    expect(mockGetQuestionStats).toBeCalledWith("easy");
  });

  it("Should show more categories on clicking `More categories`", async () => {
    const user = userEvent.setup();

    renderHomeContext(<HomePage />, {
      currentUI: {
        welcome: false,
        difficulties: true,
        categories: true,
        play: true,
      },
      showCategories: true,
    });

    const showMore = await screen.findByTestId("show-more-button");

    await user.click(showMore);

    const container = await screen.findByTestId("second-categories-set");
    expect(container).toBeInTheDocument();
  });
});
