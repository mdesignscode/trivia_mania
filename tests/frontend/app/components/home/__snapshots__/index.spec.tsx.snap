// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component Renders <Difficulties /> 1`] = `
<body>
  <div>
    <div>
      <div
        class="h-full w-full overflow-hidden"
      >
        <div
          class="col w-full h-full"
          data-testid="home-container"
        >
          <div>
            <div
              class="text-center col gap-3"
              data-testid="render-difficulties-container"
            >
              <h1
                class="text-xl"
              >
                Choose difficulty
              </h1>
              <div
                class="flex gap-2 justify-center flex-wrap"
              >
                <span
                  data-testid="easy"
                >
                  <button
                    class="sc-bcPKhP fjXGcT"
                  >
                    easy
                     (
                    3
                    )
                  </button>
                </span>
                <span
                  data-testid="hard"
                >
                  <button
                    class="sc-bcPKhP fjXGcT"
                  >
                    hard
                     (
                    2
                    )
                  </button>
                </span>
                <span
                  data-testid="medium"
                >
                  <button
                    class="sc-bcPKhP fjXGcT"
                  >
                    medium
                     (
                    3
                    )
                  </button>
                </span>
                <span
                  data-testid="all difficulties"
                >
                  <button
                    class="sc-bcPKhP fjXGcT"
                  >
                    all difficulties
                     (
                    8
                    )
                  </button>
                </span>
              </div>
            </div>
          </div>
          <div
            class="bg-accent absolute -z-10 rounded-full w-64 h-64"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`HomePage component Renders <Difficulties />, <Categories /> and <Play /> 1`] = `
<body>
  <div>
    <div>
      <div
        class="h-full w-full overflow-hidden"
      >
        <div
          class="col w-full h-full"
          data-testid="home-container"
        >
          <div
            class="flex justify-between items-center h-full"
          >
            <div
              class="col items-center justify-center bg-accent h-full rounded-full"
            >
              <div>
                <div
                  class="text-center col gap-3"
                  data-testid="render-difficulties-container"
                >
                  <h1
                    class="text-xl"
                  >
                    Choose difficulty
                  </h1>
                  <div
                    class="flex gap-2 justify-center flex-col"
                  >
                    <span
                      data-testid="easy"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        easy
                         (
                        3
                        )
                      </button>
                    </span>
                    <span
                      data-testid="hard"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        hard
                         (
                        2
                        )
                      </button>
                    </span>
                    <span
                      data-testid="medium"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        medium
                         (
                        3
                        )
                      </button>
                    </span>
                    <span
                      data-testid="all difficulties"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        all difficulties
                         (
                        8
                        )
                      </button>
                    </span>
                  </div>
                </div>
              </div>
              <div>
                <a
                  data-testid="play-link"
                  href="/game?difficulty=&categories="
                >
                  <span
                    data-testid="play-button"
                  >
                    <button
                      class="sc-bcPKhP dmjhkA"
                    >
                      Start Playing
                    </button>
                  </span>
                </a>
              </div>
            </div>
            <div
              class="text-center col gap-3 flex-1 justify-center h-11/12 overflow-y-auto"
              data-testid="render-categories-container"
            >
              <h1
                class="text-xl"
              >
                Choose categories
              </h1>
              <div
                class="col gap-3"
              >
                <div
                  class="col gap-2"
                  data-testid="display-categories-container"
                >
                  <div
                    class="flex gap-2 flex-wrap justify-center"
                    data-testid="first-categories-set"
                  >
                    <span
                      data-testid="Category 1"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 1
                         (
                        0
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 10"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 10
                         (
                        9
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 11"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 11
                         (
                        10
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 12"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 12
                         (
                        11
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 13"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 13
                         (
                        12
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 14"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 14
                         (
                        13
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 15"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 15
                         (
                        14
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 16"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 16
                         (
                        15
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 17"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 17
                         (
                        16
                        )
                      </button>
                    </span>
                    <span
                      data-testid="Category 18"
                    >
                      <button
                        class="sc-bcPKhP fjXGcT"
                      >
                        Category 18
                         (
                        17
                        )
                      </button>
                    </span>
                  </div>
                </div>
                <div
                  class="flex justify-center gap-4"
                  data-testid="display-controls-container"
                >
                  <span
                    data-testid="show-more-button"
                  >
                    <button
                      class="sc-bcPKhP coQATf"
                    >
                      More categories
                    </button>
                  </span>
                  <span
                    data-testid="reset-categories-button"
                  >
                    <button
                      class="sc-bcPKhP fjXGcT"
                    >
                      Reset categories
                    </button>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="bg-accent absolute -z-10 rounded-full  w-10/12 h-full"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`HomePage component Renders <Welcome /> 1`] = `
<body>
  <div>
    <div>
      <div
        class="h-full w-full overflow-hidden"
      >
        <div
          class="col w-full h-full"
          data-testid="home-container"
        >
          <div
            class="text-center col items-center mx-auto"
            data-testid="welcome-container"
          >
            <h1
              class="text-3xl md:text-5xl mb-3"
            >
              Welcome to Trivia Mania
            </h1>
            <p
              class="text-xl mb-5"
            >
              Test your knowledge with exciting trivia questions.
            </p>
            <span
              data-testid="welcome-continue-button"
            >
              <button
                class="sc-bcPKhP coQATf text-2xl"
              >
                Continue
              </button>
            </span>
          </div>
          <div
            class="bg-accent absolute -z-10 rounded-full w-64 h-64"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
